build-info-extractor-maven3/src/main/java/org/jfrog/build/extractor/maven/resolver/ArtifactoryEclipseResolversHelper.java

```diff
@@ -100,21 +105,79 @@ public void initResolutionRepositories(RepositorySystemSession session) {
                     builder.setProxy(proxy);
                 }
                 releaseRepository = builder.build();
-            }
-
-            List<RemoteRepository> tempRepositories = Lists.newArrayList();
-            if (releaseRepository != null) {
                 tempRepositories.add(releaseRepository);
             }
-            if (snapshotRepository != null) {
-                tempRepositories.add(snapshotRepository);
-            }
             resolutionRepositories = tempRepositories;
         }
         return resolutionRepositories;
     }
 
-    public RemoteRepository getSnapshotRepository(RepositorySystemSession session) {
+    List<ArtifactRepository> getResolutionPluginRepositories(RepositorySystemSession session) {
+        if (resolutionPluginRepositories == null) {
+            List<ArtifactRepository> tempRepositories = Lists.newArrayList();
+            initResolutionHelper(session);
+
+            String releaseRepoUrl = resolutionHelper.getRepoReleaseUrl();
+            String snapshotRepoUrl = resolutionHelper.getRepoSnapshotUrl();
+
+            org.apache.maven.artifact.repository.Authentication authentication = null;
+            if (StringUtils.isNotBlank(resolutionHelper.getRepoUsername())) {
+                authentication = new org.apache.maven.artifact.repository.Authentication(resolutionHelper.getRepoUsername(), resolutionHelper.getRepoPassword());
+            }
+            org.apache.maven.repository.Proxy proxy = null;
+            if (StringUtils.isNotBlank(resolutionHelper.getProxyHost())) {
+                proxy = new org.apache.maven.repository.Proxy();
+                proxy.setHost(resolutionHelper.getProxyHost());
+                proxy.setPort(resolutionHelper.getProxyPort());
+                proxy.setUserName(resolutionHelper.getProxyUsername());
+                proxy.setPassword(resolutionHelper.getProxyPassword());
+            }
```

Comment:
The above proxy initialisation can be probably reused, by moving it into another method. It is duplicated in the *getResolutionRepositories* method.

Reply:
It's a different proxy.
In `List<RemoteRepository> getResolutionRepositories`: org.eclipse.aether.repository
In `List<ArtifactRepository> getResolutionPluginRepositories`: org.apache.maven.repository